packages/ragbits-core/src/ragbits/core/vector_stores/_cli.py

```diff
@@ -0,0 +1,117 @@
+import asyncio
+from dataclasses import dataclass
+from pathlib import Path
+
+import typer
+from pydantic import BaseModel
+from rich.console import Console
+
+from ragbits.cli import cli_state, print_output
+from ragbits.cli.state import OutputType
+from ragbits.core.config import core_config
+from ragbits.core.embeddings.base import Embeddings
+from ragbits.core.utils.config_handling import InvalidConfigError
+from ragbits.core.vector_stores.base import VectorStore, VectorStoreOptions
+
+vector_stores_app = typer.Typer(no_args_is_help=True)
+
+
+@dataclass
+class CLIState:
+    vector_store: VectorStore | None = None
+
+
+state: CLIState = CLIState()
+
+
+@vector_stores_app.callback()
+def common_args(
+    factory_path: str | None = None,
+    yaml_path: str | None = None,
+) -> None:
+    try:
+        state.vector_store = VectorStore.subclass_from_defaults(
+            core_config,
+            factory_path_override=factory_path,
+            yaml_path_override=Path.cwd() / yaml_path if yaml_path else None,
+        )
+    except InvalidConfigError as e:
+        Console(stderr=True).print(e)
+        raise typer.Exit(1) from e
+
+
+@vector_stores_app.command(name="list")
+def list_entries(limit: int = 10, offset: int = 0) -> None:
+    """
+    List all objects in the chosen vector store.
+    """
+
+    async def run() -> None:
+        if state.vector_store is None:
+            raise ValueError("Vector store not initialized")
+
+        entries = await state.vector_store.list(limit=limit, offset=offset)
+        print_output(entries)
+
+    asyncio.run(run())
+
+
+class RemovedItem(BaseModel):
+    id: str
+
+
+@vector_stores_app.command()
+def remove(ids: list[str]) -> None:
+    """
+    Remove objects from the chosen vector store.
+    """
+
+    async def run() -> None:
+        if state.vector_store is None:
+            raise ValueError("Vector store not initialized")
+
+        await state.vector_store.remove(ids)
+        if cli_state.output_type == OutputType.text:
+            typer.echo(f"Removed entries with IDs: {', '.join(ids)}")
+        else:
+            print_output([RemovedItem(id=id) for id in ids])
+
+    asyncio.run(run())
+
+
+@vector_stores_app.command()
+def query(
+    text: str,
+    k: int = 5,
+    max_distance: float | None = None,
+    embedder_factory_path: str | None = None,
+    embedder_yaml_path: str | None = None,
+) -> None:
+    """
+    Query the chosen vector store.
+    """
+
+    async def run() -> None:
+        if state.vector_store is None:
+            raise ValueError("Vector store not initialized")
+
+        try:
+            embedder = Embeddings.subclass_from_defaults(
```

Comment:
Changing our concept of vector store embed things themselves would be quite a big change to our current model  but obviously I'm okay with you creating such a ticket. On the other hand, I'm not sure it would meaningfully help with the potential problem scenario that you describe. Adding an embedder as a field of a `VectorStore` object wouldn't "store" the information - i.e., it would be as ephemeral as before, since the objects are created from configuration each time they are used. If you have mismatched configuration, you will have a problem in both cases. If you don't - you won't have a problem in neither.

Reply:
we could accept list of texts instead of single text and define `async def run(text)`

and finally `asyncio.run(asyncio.gather(*[run(text) for text in texts]))`

i think that in this implementation it would be a very small change that would be potentially also benefitial (or tldr "low hanging fruit" :D)

Reply:
i think that to some point it would: if embedder would be part of a vector store than its name would specified within this config - if we add. eg. some embedder config hash - or even embeder name to the vector store name - than changing the embedder model would automatically also create a new vector store

Reply:
Currently, it's still hard for me to imagine - vector stores are typically external entities that we don't manage. We just connect to them based on the configuration that we get from the user. So there is no "automatically creating a new vector store" for us. It's like connecting to a remote SQL database. 

But if you have an idea how to do this, create a ticket and we will talk about it during grooming.